src/vicsetup/libvicsetup/luks.c

```diff
@@ -0,0 +1,585 @@
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <sys/stat.h>
+#include <limits.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "vic.h"
+#include "lukscommon.h"
+#include "byteorder.h"
+#include "luks2.h"
+#include "luks1.h"
+#include "raise.h"
+#include "hexdump.h"
+#include "integrity.h"
+#include "dm.h"
+#include "strings.h"
+
+VIC_STATIC_ASSERT(VIC_OFFSETOF(vic_luks_hdr_t, magic) == 0);
```

Comment:
Can we have a comment explaining why these fields need to be at these offsets? (Is this matching an on-disk structure?)